packages/ngtools/webpack/src/transformers/replace_resources.ts

```diff
@@ -327,3 +385,14 @@ function getDecoratorOrigin(
 
   return null;
 }
+
+/** Checks if the current version of TypeScript is after the specified major/minor versions. */
+function isAfterVersion(targetMajor: number, targetMinor: number): boolean {
+  const [major, minor] = ts.versionMajorMinor.split('.').map((part) => parseInt(part));
+
+  if (major < targetMajor) {
+    return false;
+  }
+
+  return major === targetMajor ? minor >= targetMinor : true;
```

Comment:
Nit: I recommend leaving this logic in the if/else for symmetry. Otherwise the `major > targetMajor` case is a bit obfuscated.

```typescript
if (major < targetMajor) {
  return false;
} else if (major > targetMajor) {
  return true;
} else {
  return minor => targetMinor;
}
```